haptools/sim_genotype.py

```diff
@@ -231,7 +280,30 @@ def _convert_haplotype(haplotype, chrom, pop_dict, pop_sample, sample_dict):
     return np.asarray(hap_pos, dtype=np.int64), \
            np.asarray(hap_pops, dtype=np.uint8), \
            np.asarray(hap_samples, dtype=object), \
-           np.asarray(hap_samples_ind, dtype=np.int32)
+           np.asarray(hap_samples_ind, dtype=np.int32), \
+           np.asarray(hap_inds, dtype=np.uint8)
+
+def _find_random_sample(samples, sample_dict, haps_used, chrom, start_coord, end_coord):
+    for sample in samples:
```

Comment:
I know that `samples` (which is `pop_sample[population]`) was randomized at the beginning. Is it the case that, for example, the first individual simulated is more likely to copy from the first haplotype in `samples`? Might want to ensure that we don't end up copying mostly from one or a small number of samples for any given simulated haplotype.